### PR TITLE
Improving SCE GU readability

### PR DESCRIPTION
**Pull request by fjtrujy**
## Description
This PR isn't changing anything regarding performance, binary size, or public API.

It just improves readability, replacing "magic numbers" with something meaningful, I hope this will help to understand better how things work under the hood.

I will create separate PRs where I will try to improve function inlining making it to perform faster.

Please review it carefully as I could have had some mistakes as there are tons of files changes.

Cheers.

### CODE REVIEW

File: src/gu/sceGuDisable.c [hunk 1]
```diff
@@ -10,43 +10,83 @@
...
-			sendCommandi(201,context->texture_function);
-		}
+	case GU_ALPHA_TEST:
+		sendCommandi(ALPHA_TEST_ENABLE, 0);
 		break;
+	case GU_DEPTH_TEST:
+		sendCommandi(Z_TEST_ENABLE, 0);
+		break;
+	case GU_SCISSOR_TEST:
+	{
```
Comment (sharkwouter) [inline src/gu/sceGuDisable.c:22]: The indentationhere looks off to me

File: src/gu/guInternal.h [hunk 2]
```diff
@@ -17,113 +17,382 @@ typedef struct
...
-void callbackSig(int id, void* arg);
-void callbackFin(int id, void* arg);
+void callbackSig(int id, void *arg);
+void callbackFin(int id, void *arg);
 void resetValues();
 
-static __inline__ void sendCommandi(int cmd, int argument)
+typedef enum GECommand
+{
+	NOP = 0x0,			/* No operation */
```
Comment (sajattack) [inline src/gu/guInternal.h:123]: maybe some doxygen syntax sugar on these variants would be good so these comments don't go to waste.

**Review by sajattack — APPROVED** [review]

File: src/gu/guInternal.h (hunk 2 above)
Comment (fjtrujy) [reply src/gu/guInternal.h:123]: do you have some documentation about how to do it?

Comment (sharkwouter) [reply src/gu/guInternal.h:123]: You put the comment with the description 1 line above the enum entry.

File: src/gu/sceGuAmbient.c [hunk 3]
```diff
@@ -10,6 +10,6 @@
 
 void sceGuAmbient(unsigned int color)
 {
-	sendCommandi(92,(color & 0xffffff));
-	sendCommandi(93,(color >> 24));
+	sendCommandi(AMBIENT_COLOR,(color & 0xffffff));
```
Comment (sharkwouter) [inline src/gu/sceGuAmbient.c:13]: These 2 are wrong. AMBIENT_COLOR is  85, AMBIENT_ALPHA is 88.

File: src/gu/sceGuAmbientColor.c [hunk 4]
```diff
@@ -10,6 +10,6 @@
 
 void sceGuAmbientColor(unsigned int color)
 {
-	sendCommandi(85,color & 0xffffff);
-	sendCommandi(88,color >> 24);
+	sendCommandi(AMBIENT_LIGHT_COLOR, color & 0xffffff);
```
Comment (sharkwouter) [inline src/gu/sceGuAmbientColor.c:13]: I think AMBIENT_LIGHT_COLOR and AMBIENT_COLOR got switched, as well as AMBIENT_ALPHA and AMBIENT_LIGHT_ALPHA. They are 92 and 93 here now.

File: src/gu/sceGuEndObject.c [hunk 5]
```diff
@@ -12,11 +12,11 @@ void sceGuEndObject(void)
...
 
-	unsigned int* current = gu_list->current;
-	gu_list->current = gu_object_stack[gu_object_stack_depth-1];
+	unsigned int *current = gu_list->current;
+	gu_list->current = gu_object_stack[gu_object_stack_depth - 1];
 
-	sendCommandi(16, (((unsigned int)current) >> 8) & 0xf0000);
-	sendCommandi(9, ((unsigned int)current) & 0xffffff);
+	sendCommandi(BASE, (((unsigned int)current) >> 8) & 0xf0000);
+	sendCommandi(END, ((unsigned int)current) & 0xffffff);
```
Comment (sharkwouter) [inline src/gu/sceGuEndObject.c:19]: This one does not match the original value. END is 12.

File: src/gu/sceGuMaterial.c [hunk 6]
```diff
@@ -12,13 +12,13 @@ void sceGuMaterial(int mode, int color)
 {
 	if (mode & 0x01)
 	{
-		sendCommandi(85, color & 0xffffff);
-		sendCommandi(88, color >> 24);
+		sendCommandi(AMBIENT_COLOR, color & 0xffffff);
```
Comment (sharkwouter) [inline src/gu/sceGuMaterial.c:15]: Here AMBIENT_COLOR and AMBIENT_ALPHA have the correct values.

File: src/gu/sceGuViewport.c [hunk 7]
```diff
@@ -10,8 +10,8 @@
 
 void sceGuViewport(int cx, int cy, int width, int height)
 {
-	sendCommandf(66,(float)(width>>1));
-	sendCommandf(67,(float)((-height)>>1));
-	sendCommandf(69,(float)cx);
-	sendCommandf(70,(float)cy);
+	sendCommandf(VIEWPORT_X_SCALE, (float)(width >> 1));
+	sendCommandf(VIEWPORT_X_SCALE, (float)((-height) >> 1));
```
Comment (sharkwouter) [inline src/gu/sceGuViewport.c:14]: This one should be 67, but VIEWPORT_X_SCALE matches 66.

File: src/gu/sceGuInit.c
Comment (sharkwouter) [inline src/gu/sceGuInit.c:1]: I'm going to need more time to be able to verify this one.

File: src/gu/guInternal.h (hunk 2 above)
Comment (sajattack) [reply src/gu/guInternal.h:123]: https://stackoverflow.com/questions/1717941/issue-getting-doxygen-to-document-an-enum-in-c

File: src/gu/sceGuAmbient.c (hunk 3 above)
Comment (fjtrujy) [reply src/gu/sceGuAmbient.c:13]: Done!

File: src/gu/sceGuEndObject.c (hunk 5 above)
Comment (fjtrujy) [reply src/gu/sceGuEndObject.c:19]: Done!

File: src/gu/sceGuAmbientColor.c (hunk 4 above)
Comment (fjtrujy) [reply src/gu/sceGuAmbientColor.c:13]: Done!

File: src/gu/sceGuEndObject.c (hunk 5 above)
Comment (fjtrujy) [reply src/gu/sceGuEndObject.c:19]: Thanks! Fixed

File: src/gu/sceGuMaterial.c (hunk 6 above)
Comment (fjtrujy) [reply src/gu/sceGuMaterial.c:15]: 🆗

File: src/gu/sceGuViewport.c (hunk 7 above)
Comment (fjtrujy) [reply src/gu/sceGuViewport.c:14]: Good catch!

File: src/gu/sceGuInit.c [hunk 8]
```diff
@@ -12,39 +12,266 @@
...
+		ZV(MORPH_WEIGHT2),
+		ZV(MORPH_WEIGHT3),
+		ZV(MORPH_WEIGHT4),
+
+		ZV(MORPH_WEIGHT5),
+		ZV(MORPH_WEIGHT6),
+		ZV(MORPH_WEIGHT7),
+		ZV(PATCH_DIVISION),
+		ZV(PATCH_FACING),
+		ZV(PATCH_PRIMITIVE),
```
Comment (sharkwouter) [inline src/gu/sceGuInit.c:64]: These 2 need to be flipped around. They are in each other's place in the array

File: src/gu/sceGuInit.c [hunk 9]
```diff
@@ -12,39 +12,266 @@
...
+		ZV(VERTEX_WEIGHT_SIZE),
+		ZV(VERTEX_WEIGHT_TYPE),
+		ZV(VERTEX_WEIGHT_MODEL),
+
+		ZV(VERTEX_WEIGHT_NORMALIZE),
+		ZV(VERTEX_WEIGHT_OFFSET),
+		ZV(VERTEX_WEIGHT_SCALE),
+
+		ZV(TRANSFER_OUT_SIZE),
+		ZV(TEX_MAP_MODE),
```
Comment (sharkwouter) [inline src/gu/sceGuInit.c:271]:
```suggestion
		ZV(FINISH),
		ZV(END),
```
Like this these match.

File: src/gu/sceGuInit.c [hunk 10]
```diff
@@ -12,39 +12,266 @@
...
+		ZV(DITH0),
+		ZV(DITH1),
+
+		ZV(DITH2),
+		ZV(DITH3),
+		ZV(LOGIC_OP),
+		ZV(Z_MASK),
+		ZV(MASK_COLOR),
+		ZV(MASK_ALPHA),
+		ZV(TRANSFER_START),
```
Comment (sharkwouter) [inline src/gu/sceGuInit.c:254]: This should be TRANSFER_SRC_OFFSET

File: src/gu/sceGuInit.c [hunk 11]
```diff
@@ -12,39 +12,266 @@
...
+		ZV(TEX_ADDR0),
+
+		ZV(TEX_ADDR1),
+		ZV(TEX_ADDR2),
+		ZV(TEX_ADDR3),
+		ZV(TEX_ADDR4),
+		ZV(TEX_ADDR5),
+		ZV(TEX_ADDR6),
+		ZV(TEX_ADDR7),
+		CV(TEX_BUF_WIDTH0, 512 * 512),
```
Comment (sharkwouter) [inline src/gu/sceGuInit.c:183]:
```suggestion
		CV(TEX_BUF_WIDTH0, 512 * 512) + 4,
```
I don't understand this, but like this this one matches. I don't know if there is some bug here or something in the original.

File: src/gu/sceGuInit.c [hunk 12]
```diff
@@ -12,39 +12,266 @@
...
+		ZV(TEX_BUF_WIDTH6),
+		ZV(TEX_BUF_WIDTH7),
+		ZV(CLUT_ADDR),
+
+		ZV(CLUT_ADDR_UPPER),
+		ZV(TRANSFER_SRC),
+		ZV(TRANSFER_SRC_W),
+		ZV(TRANSFER_DST),
+		ZV(TRANSFER_DST_W),
+		CV(TEX_SIZE0, 257),
```
Comment (sharkwouter) [inline src/gu/sceGuInit.c:199]: Where does this 257 come from?

Comment (sharkwouter) [reply src/gu/sceGuInit.c:199]: Maybe 0x101 would look a bit cleaner. Still a magic number, though.

File: src/gu/sceGuInit.c (hunk 8 above)
Comment (fjtrujy) [reply src/gu/sceGuInit.c:64]: Done!

File: src/gu/sceGuInit.c (hunk 9 above)
Comment (fjtrujy) [reply src/gu/sceGuInit.c:271]: Done!

File: src/gu/sceGuInit.c (hunk 10 above)
Comment (fjtrujy) [reply src/gu/sceGuInit.c]: Done!

File: src/gu/sceGuInit.c (hunk 11 above)
Comment (fjtrujy) [reply src/gu/sceGuInit.c]: Done!

**Comment by rofl0r** [comment]
fwiw, the correctness of changes like these (where only constants are being replaced) can easily be verified by doing a before-after comparison of the object files with tools like vbindiff or even md5sum. if the conversion is correct, then the object files should be identical.

**Comment by sharkwouter** [comment]
I wasn't aware. I looked at the enum values in an ide to make sure they were correct and wrote a small program to compare the old and the new init arrays.

**Comment by fjtrujy** [comment]
> fwiw, the correctness of changes like these (where only constants are being replaced) can easily be verified by doing a before-after comparison of the object files with tools like vbindiff or even md5sum. if the conversion is correct, then the object files should be identical.

Well I have modified some `switch`s `if`s so maybe the generated object is not 100% the same

**Pull request merged**